src/definitions/modules/slider.js

```diff
@@ -193,48 +201,39 @@ $.fn.slider = function(parameters) {
             var
               $children   = $labels.find('.label'),
               numChildren = $children.length,
-              ratio,
-              position
+              min         = module.get.min(),
+              max         = module.get.max(),
+              ratio
             ;
             $children.each(function(index) {
               var
-                $child = $(this),
+                $child    = $(this),
                 attrValue = $child.attr('data-value')
               ;
               if(attrValue) {
-                position = module.determine.positionFromValue(attrValue)
+                attrValue = attrValue > max ? max : attrValue < min ? min : attrValue;
+                ratio = (attrValue - min) / (max - min);
               } else {
-                ratio = ((index+1)/(numChildren+1));
-                position = module.determine.positionFromRatio(ratio);
+                ratio = (index + 1) / (numChildren + 1);
               }
-              var posDir =
-                module.is.vertical()
-                ?
-                module.is.reversed() ? 'bottom' : 'top'
-                :
-                module.is.reversed() ? 'right' : 'left'
-              ;
-              $(this).css(posDir, position);
+              module.update.labelPosition(ratio, $(this));
             });
           },
           autoLabel: function() {
             if(module.get.step() != 0) {
               $labels = $module.find('.labels');
-              if($labels.length != 0)
-                $labels.empty()
-              else
+              if($labels.length != 0) {
+                $labels.empty();
+              }
+              else {
                 $labels = $module.append('<ul class="auto labels"></ul>').find('.labels');
-              for(var i = 0; i <= module.get.numLabels(); i++) {
+              }
+              for(var i = 1, len = module.get.numLabels(); i <= len; i++) {
                 var
-                  $label = $('<li class="label">' + module.get.label(i+1) + '</li>'),
-                  position =
-                    module.is.vertical()
-                    ?
-                    module.is.reversed() ? 'bottom' : 'top'
-                    :
-                    module.is.reversed() ? 'right' : 'left'
+                  $label = $('<li class="label">' + module.get.label(i) + '</li>'),
+                  ratio  = i / (len + 1)
```

Comment:
this isn't quite right. If I have a max of 10 and a min of 0, I am going to have 8 labels. So this would be `1/9` not `1/10` as it should be.

With the change to `module.get.numLables()`: `i / (len + 1)` -> `i / len`